strawberry/field.py

```diff
@@ -246,6 +247,13 @@ def get_result(
         if self.base_resolver:
             return self.base_resolver(*args, **kwargs)
 
+        if inspect.isawaitable(source):
```

Comment:
Is this necessary? When would this be asynchronous?

Reply:
In a nutshell: Our tests pass without it but the following schema would produce an error when queried using the debug server

```python
import strawberry


class CanView(strawberry.BasePermission):
    async def has_permission(self, source, info, **kwargs):
        return True


@strawberry.type
class User:
    name: str


@strawberry.type
class Query:
    @strawberry.field(permission_classes=[CanView])
    async def user(self, name: str) -> User:
        return User(name=name)


schema = strawberry.Schema(query=Query)
```

I'm currently further investigating this issue...

Reply:
Found the issue, in some cases unresolved results were returned from the async resolver function I added. Didn't noticed this was an issue because I missinterpreted some of the graphql-core code. I updated the test to cover this case and fixed it.